pygeos/decorators.py

```diff
@@ -16,19 +17,33 @@ def __init__(self, version):
         self.version = tuple(int(x) for x in version.split("."))
 
     def __call__(self, func):
-        if lib.geos_version < self.version:
-            msg = "'{}' requires at least GEOS {}.{}.{}".format(
-                func.__name__, *self.version
-            )
+        msg = "'{}' requires at least GEOS {}.{}.{}.".format(
+            func.__name__, *self.version
+        )
+        if os.environ.get("SPHINX_DOC_BUILD") == "1":  # set in docs/conf.py
+            # Doc build: just add a note about the version constraint after
+            # the first line of the docstring.
+            @wraps(func)
+            def wrapped(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            if wrapped.__doc__:
+                wrapped.__doc__ = wrapped.__doc__.replace(
+                    "\n\n", "\n\n    .. note:: {}\n\n".format(msg), 1
```

Comment:
I would personally do this for the case of `lib.geos_version < self.version` as well. Also if your geos version is too old, it can be useful to be able to read the docstring